src/createKindeClient.ts

```diff
@@ -99,23 +101,48 @@ const createKindeClient = async (
   const setStore = (data: KindeState & {error: string}) => {
     if (!data || data.error) return;
 
-    const accessToken = jwtDecode(data.access_token);
     const idToken = jwtDecode(data.id_token)! as JWT & KindeUser;
-    store.setItem('kinde_token', data);
-    store.setItem('kinde_access_token', accessToken);
-    store.setItem('kinde_id_token', idToken);
-    store.setItem('user', {
-      id: idToken.sub,
-      given_name: idToken.given_name,
-      family_name: idToken.family_name,
-      email: idToken.email,
-      picture: idToken.picture
-    });
+    const idTokenHeader = jwtDecode(data.id_token, {header: true});
+    const accessToken = jwtDecode(data.access_token);
+    const accessTokenHeader = jwtDecode(data.access_token, {header: true});
 
-    if (is_dangerously_use_local_storage) {
-      localStorage.setItem('kinde_refresh_token', data.refresh_token);
-    } else {
-      store.setItem('kinde_refresh_token', data.refresh_token);
+    const validatorOptions = {
+      iss: domain,
+      azp: clientId,
+      aud: audience
+    };
+    const isIDValid = isTokenValid(
+      {
+        payload: idToken,
+        header: idTokenHeader
+      },
+      {...validatorOptions, aud: clientId}
+    );
+    const isAccessValid = isTokenValid(
+      {
+        payload: accessToken,
+        header: accessTokenHeader
+      },
+      validatorOptions
+    );
+
+    if (isIDValid && isAccessValid) {
+      store.setItem('kinde_token', data);
+      store.setItem('kinde_access_token', accessToken);
+      store.setItem('kinde_id_token', idToken);
+      store.setItem('user', {
+        id: idToken.sub,
+        given_name: idToken.given_name,
+        family_name: idToken.family_name,
+        email: idToken.email,
+        picture: idToken.picture
+      });
+
+      if (is_dangerously_use_local_storage) {
+        localStorage.setItem('kinde_refresh_token', data.refresh_token);
+      } else {
+        store.setItem('kinde_refresh_token', data.refresh_token);
+      }
```

Comment:
Instead of using strings here, I think we should use an enum for the item keys

Reply:
Updated, thanks :)